### PR TITLE
Remove "free-for-all" memory allocation for bundles and workers

### LINKED ISSUE #871: Get rid of "free-for-all" memory allocation for run bundles
From now on, just assign a default memory limit to bundles.

### PR DESCRIPTION
**Pull request by fabeschan**
Fixes #871

### CODE REVIEW

**Comment by fabeschan** [comment]
Do not merge this PR until bundle request metadata has non-NoneType default values

Do not merge this PR until bundle request memory metadata has non-NoneType and positive default value

**Review by percyliang — APPROVED** [review]

**Comment by fabeschan** [comment]
Tests are broken right now because bundle request memory defaults to 0. Once changes to the default value are merged, rebase this PR and tests should pass.

**Pull request merged**
